File: src/lib/events.ts
import { listen, type UnlistenFn } from "@tauri-apps/api/event";

/** Event names emitted by the Rust/Tauri boundary. Keep this list in sync with the backend emitters. */
export const dropEvents = {
  peersUpdated: "peers-updated",
  transferUpdate: "transfer-update",
  incomingTransfer: "incoming-transfer",
  trustRequest: "trust-request",
  discoveryStatus: "discovery-status",
  connectivityDiagnostics: "connectivity-diagnostics",
} as const;

export type DropEventName = (typeof dropEvents)[keyof typeof dropEvents];

export function subscribeDropEvent<T>(
  event: DropEventName,
  handler: (payload: T) => void,
): Promise<UnlistenFn> {
  return listen<T>(event, ({ payload }) => handler(payload));
}

File: src-tauri/src/events.rs
//! Names for the Rust-to-frontend event boundary.
//!
//! Payload types are defined by the emitting subsystem; keeping the names in
//! one place prevents a UI rename from silently leaving a stale emitter.

pub(crate) const PEERS_UPDATED: &str = "peers-updated";
pub(crate) const TRANSFER_UPDATE: &str = "transfer-update";
pub(crate) const INCOMING_TRANSFER: &str = "incoming-transfer";
pub(crate) const TRUST_REQUEST: &str = "trust-request";
pub(crate) const DISCOVERY_STATUS: &str = "discovery-status";
pub(crate) const CONNECTIVITY_DIAGNOSTICS: &str = "connectivity-diagnostics";
